### ISSUE TITLE
video upload: progress bar, satellite and OSM map layer lacking

### ISSUE THREAD

**Issue by ligfietser** [mapillary/mapillary_issues#881]
I'm trying out the new video feature, it's great to see this is added to Mapillary!
However, after I pressed upload video, no progress bar is visible, just a white screen with a blinking cursor on the right side. I use Firefox 36.0.4 / W7 64bits *. I dont have a clue that something is being uploaded to the server (I guess not?)

Another issue is that it is hard to see any map items in the default mapbox map. Better add a satellite layer and OSM layer too.

*) Google Chrome 41.0.2272.101, IE 11.0 also not working.

**Comment by ligfietser** [comment]
Update: Progress bar is now visible, but it seems to get stuck in uploading... after the green progress bar has reached the 100%, no high five.

**Comment by ybkuang** [comment]
@ligfietser Thanks for testing this out.  Can you check whether there are errors in the console and paste them here?

**Comment by ybkuang** [comment]
I just checked that your gpx file went through to our server just now, but the video is missing. So something went wrong there.

**Comment by ligfietser** [comment]
@ybkuang I didnt see any errors. Here you can test my video file, maybe your scripts can't handle/recognize the format? I have converted it from mov to mp4. https://drive.google.com/file/d/0B5LoKTN293hkb1l0TWJUYlNtcjQ/view?usp=sharing

**Comment by ybkuang** [comment]
The video file did not reach our server. Could be a problem when transferring the file. I will take a look at your video.

**Comment by gyllen** [comment]
We had a hard limit of 200MB for the video upload. I bumped that to 1GB, try uploading again.

**Comment by gyllen** [comment]
By the way @ligfietser did you notice I added other layers than the mapbox tile layer.

**Comment by ligfietser** [comment]
Seems to work now, thanks!
@gyllen no, I still see only one mapbox layer.

Another issue: I have to select my gpx file twice before the map is loading.

**Comment by gyllen** [comment]
@ligfietser Hmm might be that I just changed for images, @knikel can you fix other maps for video as well?

**Comment by knikel** [comment]
Yes, on it.

**Comment by gyllen** [comment]
@ligfietser Double checked and we have now received your video file. You need to hang on for a while though we are currently carefully importing all files one by one.

**Comment by ligfietser** [comment]
Thanks guys, I also noticed the satellite and OSM layer now, makes it a lot easier to allign the video with the GPX file. 

A small issue that is left: if you select a gpx file, nothing happens. You must select it again before the map appears.

**Comment by ToeBee** [comment]
Is there still a 1 GB file limit? It should check that before starting the upload and say something. The files from my GoPro camera are all ~2 GB so I will have to split them up if you limit file size like this.

**Comment by gyllen** [comment]
I can kick the limit up to 2.5GB

**Comment by ToeBee** [comment]
That would work. I think the GoPro actually splits the video file when it hits 2 GB if you are continuously shooting.

**Comment by gyllen** [comment]
Ok I just switched. Reload page and retry upload.

**Comment by gyllen** [comment]
@ToeBee I see an upload from you but no video file. Did upload fail?

**Comment by ToeBee** [comment]
Yes, it stalled out at 30 or 40%. This happened last night from home as well although at that point I guess the video file was over your size limit anyway. I brought the files to work with me where I have a 100 Mbit pipe thinking a faster connection would improve things but no such luck. Kind of annoying that I have to re-sync the position every time. The video I'm trying to upload is from western Kansas where there aren't a lot of landmarks to sync up position so I have to seek a couple minutes into the video :)

#ruralproblems

**Comment by gyllen** [comment]
@ToeBee Damn does it still stop at 30-40%?

**Comment by gyllen** [comment]
Sorry silly question. I misread your report.

**Comment by gyllen** [comment]
Do you have anyway to give me access to your file so I can debug?

**Comment by ToeBee** [comment]
This time it stopped at like 10%. On my first attempt last night the progress bar appeared to be at 100% but it didn't actually finish. 

I tossed the file up to dropbox:
https://dl.dropboxusercontent.com/u/1475575/mapillary/GOPR0001.MP4
If you need the GPX file too, 
https://dl.dropboxusercontent.com/u/1475575/mapillary/activity_324365637.gpx

**Comment by gyllen** [comment]
Downloaded video and trying it out. I added a % counter to web. Would you mind shift+reload page and try an upload again.

**Comment by ToeBee** [comment]
Worked that time! Also, wow that road was rough that day. The camera is shaking all over the place.

**Comment by gyllen** [comment]
Yeah I saw that as well, but I think the stills will still be good. Really just noticing the rolling shutter when played as video. Lets see if we can import the files (we are closely watching every import now to iron out any possible bugs)

**Comment by ToeBee** [comment]
Hah. I just got a nastygram from Dropbox that my public links are causing "excessive traffic" and have been disabled. The previous links will be disabled :)

**Comment by gyllen** [comment]
:)

**Comment by gyllen** [comment]
Images are importing into system (see "still processing" here http://www.mapillary.com/profile/toebee). Gonna take a while before they are visible though due to heavy upload the last couple of hours.

**Comment by ToeBee** [comment]
I got an email that the upload has been registered. Out of curiosity, how are you determining how often to grab a frame? It grabbed just over 3,000 pictures from that 26 minute video so it looks like you may just be doing a flat 2 frames per second. Seems a little high for a video of western Kansas grasslands, shot at bicycle speeds. I suppose in a more urban environment that rate may make more sense. Anyway, I was just curious if you are attempting to vary the frame rate based on speed or anything fancy like that.

**Comment by gyllen** [comment]
@ybkuang Actually implemented this, but I think it is sampled every 3 meters, which might be a little to tight even in urban areas. In Kansas all we need is one picture right :)

**Comment by ToeBee** [comment]
Hey now! Although looking at that video again... yeah that's pretty accurate for that day at least. I promise, there's a town coming up eventually.

Speaking of which, I just tried uploading another one and had to attempt it 3 times before it went through without freezing up. The first time at 15% and the next time at 80-something%

Also, I found out I can reliably crash my Chromium tab by attempting to select an MP4 file to upload in image mode. Guess that might be more of a Chromium bug really.

**Comment by ybkuang** [comment]
@ToeBee Awesome with the video and can't wait to see the town ;) As @gyllen mentioned, the images are sampled every 3 meters from the video. Video upload is new to us, as uploads come along, we will understand what sampling distance is better.

**Comment by ligfietser** [comment]
Thanks for the fixes, the percentage of the progress bar is cool.
I still have the issue that selecting the gpx file seems to do nothing. I have to select it again before the map pops up. Can you have a look at this?

**Comment by knikel** [comment]
@ligfietser The issue with selecting again the .gpx file should be fixed now.

**Comment by ligfietser** [comment]
@knikel: it is fixed in Chrome, unfortunately this is still not fixed in FF (37.0.1) or IE 11, 

Another issue, I uploaded a video 3 days ago (MAH00845.MP4) but it is still not added to the map (my activity shows 362 photos uploaded 17 hrs ago but nothing seems to happen after this).

**Comment by ToeBee** [comment]
So the upload stalling out is still a problem. Tonight after work I stayed here a while (to use the fast .edu connection to test) and tried to upload a video 3 or 4 times and it failed every time. Then I fired up wireshark to capture packets and see if anything weird stood out there. But ever since I started wireshark, my uploads have been going through without fail. I've uploaded 4 videos in a row now. So that's weird. Are others still seeing hanging uploads?

**Comment by gyllen** [comment]
I think its a network issue. We should really provide a tool for uploading videos. Uploads actually never touch our (mapillary) servers it goes straight to Amazon Web Services. My guess is that it is a break in connection and the browser is not able to recover from that.

**Comment by gyllen** [comment]
@ToeBee Im clsoing this one since I think it is a network issue. Please reopen if you feel otherwise. Also note that we are looking on other solutions than just uploading from web.
